### PR TITLE
Gate service

### PR DESCRIPTION
**Pull request by Obult**
added authentication for the chat gateway based on JWT at connection establishment

### CODE REVIEW

**Review by LJmartin94 — APPROVED** [review]
It looks great, just please don't shoot. I have so much left to lint for.

File: backend/volume/src/cookie/cookie.service.ts
Comment (rvan-duy) [inline backend/volume/src/cookie/cookie.service.ts:1]: Same for the services

File: backend/volume/src/cookie/cookie.module.ts
Comment (rvan-duy) [inline backend/volume/src/cookie/cookie.module.ts:1]: Is this module used? If not please remove it.

File: backend/volume/src/user/prisma/prismaUser.service.ts
Comment (rvan-duy) [inline backend/volume/src/user/prisma/prismaUser.service.ts:1]: `UserChats` should be `userChats`

File: backend/volume/src/chat/chat.gateway.ts [hunk 1]
```diff
@@ -23,101 +26,194 @@ export class ChatGateway implements OnGatewayInit, OnGatewayConnection, OnGatewa
...
+  async handleConnection(client: Socket) {
+    if (client.handshake.auth.token === '')
+      return;
+    const user = await this.jwtService.verify(
+      client.handshake.auth.token, { secret: process.env.JWT_SECRET }
+    );
+    // if error maybe we need to recover or exit or something!!!
 
-    client.emit('loadAllChats'); // data, all chats roomDto[]
+    // waarom is id 'sub' ???
```
Comment (rvan-duy) [inline backend/volume/src/chat/chat.gateway.ts:70]: `.verify()` returns a payload object, not a user object. This is also why `sub` contains the user id. As it is the Subject of the payload.

**Review by rvan-duy — CHANGES_REQUESTED** [review]
Made some comments, have a look at them.

**Comment by LJmartin94** [comment]
I implemented the changes you requested at @rvan-duy  <3

**Review by rvan-duy — APPROVED** [review]

**Pull request merged**